lib/commands/general.js

```diff
@@ -49,15 +49,14 @@ commands.getWindowSize = async function (windowHandle = 'current') {
 };
 
 commands.hideKeyboard = async function (strategy, ...possibleKeys) {
-  let keyboard;
   try {
-    keyboard = await this.findNativeElementOrElements('class name', 'XCUIElementTypeKeyboard', false);
-  } catch (err) {
-    // no keyboard found
-    log.debug('No keyboard found. Unable to hide.');
+    await this.proxyCommand('/wda/keyboard/dismiss', 'POST');
     return;
+  } catch (err) {
+    log.debug('Cannot dismiss the keyboard using the native call. Trying to apply a workaround...');
   }
 
+  const keyboard = await this.findNativeElementOrElements('class name', 'XCUIElementTypeKeyboard', false);
```

Comment:
it is done intentionally, since keyboard presence verification is already done by WDA's _/wda/keyboard/dismiss_ method and it only throws an error if the keyboard is still present.